services/py-api/src/server/handlers/hackathon_handlers.py

```diff
@@ -40,4 +41,5 @@ async def delete_participant(
             response.status_code = status.HTTP_500_INTERNAL_SERVER_ERROR
             return ErrResponse(error="An unexpected error occurred during the deletion of Participant")
 
+        response.status_code = status.HTTP_200_OK
```

Comment:
Why is the delete handler in hackathon but the create one is in participants?

Reply:
Alright, so since the Hackathon Service is the only service that direct access to the repositories as for the current structure, we were kinda forced to write the delete logic there, because every other implementation would either break encapsulation or wasn't what we were looking for in terms of structure. This way, we decided to keep it for all the hackathon management business logic (which would be utilized by an admin user), while the creation of a participant is mainly utilized by the clients (people who want to register). We are aware that this might not be ideal, its one of the few solutions that we came up with. 
I am open to suggestions if you have any. Thank you Misho!

Reply:
Why do you have the participant service then?